crates/transaction-pool/src/pool/mod.rs

```diff
@@ -1096,3 +1096,92 @@ impl<T: PoolTransaction> OnNewCanonicalStateOutcome<T> {
         FullPendingTransactionIter { kind, iter }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        blobstore::{BlobStore, InMemoryBlobStore},
+        test_utils::{MockTransaction, TestPoolBuilder},
+        validate::ValidTransaction,
+        BlockInfo, PoolConfig, SubPoolLimit, TransactionOrigin, TransactionValidationOutcome, U256,
+    };
+    use reth_primitives::{kzg::Blob, transaction::generate_blob_sidecar};
+    use std::{fs, path::PathBuf};
+
+    #[test]
+    fn test_discard_blobs_on_blob_tx_eviction() {
+        // Define the maximum limit for blobs in the sub-pool.
+        let blob_limit = SubPoolLimit::new(1000, usize::MAX);
+
+        // Create a test pool builder with default configuration and the specified blob limit.
+        let pool_builder =
+            TestPoolBuilder::default().with_config(PoolConfig { blob_limit, ..Default::default() });
+
+        // Set the block info for the pool, including a pending blob fee.
+        pool_builder
+            .pool
+            .pool
+            .write()
+            .set_block_info(BlockInfo { pending_blob_fee: Some(10_000_000), ..Default::default() });
```

Comment:
might be able to use `TransactionPoolExt` to call this method without calling `read`, same with other pool methods

Reply:
Done